### PR TITLE
Add documentation for connecting to Azure

### PR DESCRIPTION
**Pull request by ollietreend**
I've documented the process of connecting to Azure. This is intended as a bit of a 'cheat sheet' for us to reference quickly without having to delve into the 'real' docs whenever we need to open a Rails console.

## Formatted view ✨ 

https://github.com/DFE-Digital/itt-mentor-services/blob/document-rails-console/docs/connect-to-azure.md

## Updated `.tool-versions`

I've also updated the `.tool-versions` file so developers can easily install the required tooling for connecting to Azure.

### CODE REVIEW

**Review by JamieCleare2525 — APPROVED** [review]
Nice documentation.

File: .tool-versions [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+ruby      3.2.2
+yarn      1.22.1
+
+# Cloud infrastructure dependencies
+# https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/developer-onboarding.md#software-requirements
+azure-cli 2.56.0
+jq        1.7.1
+kubectl   1.29.1
+python    3.10.11 # required by azure-cli
+terraform 1.6.4 # should match terraform_version in .github/actions/deploy-environment/action.yml
```
Comment (ollietreend) [inline .tool-versions:12]: Update readme to include `asdf plugin add` commands for these plugins.

File: docs/connect-to-azure.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+   >
+   > Click the "Activate" button at the bottom of the pane.
+   >
+   > ![](images/connect-to-azure/azure-resource-activation.png)
+
+5. Open a console and run:
+
+   ```
+   $ cd ~/path/to/itt-mentor-services
+   $ make qa get-cluster-credentials
```
Comment (ollietreend) [inline docs/connect-to-azure.md:44]:
```suggestion
   $ cd ~/path/to/itt-mentor-services
   $ az login
   $ make qa get-cluster-credentials
```

Comment (ollietreend) [reply docs/connect-to-azure.md:44]: Ignore the error:

```
make: kubelogin: No such file or directory
make: *** [get-cluster-credentials] Error 1
```

it if happens

**Review by Nitemaeric — APPROVED** [review]

**Pull request merged**